zero_bin/leader/src/client.rs

```diff
@@ -33,7 +33,8 @@ pub struct ProofParams {
 
 /// The main function for the client.
 pub(crate) async fn client_main(
-    runtime: Runtime,
+    block_proof_runtime: Runtime,
+    segment_proof_runtime: Runtime,
```

Comment:
Maybe we could create some `ProofRuntime` structure that contains both arguments and pass them as one parameter through the stack?

Reply:
fixed